## Task

Our (hypothetical) marketing department needs some help in focussing on the most important market segment. Instead of marketing to everyone, they need our help in identifying who our most profitable customers are. Are they married or unmarried, working or retired, premium or budget buyers etc… This will help them position things better.

Also, they need help in identifying the bestselling products. 

So, given the transaction data, help us identify the top 3 most profitable products and characteristics of our most loyal customers.

- married,unmariied,working or retired ,premiume or budget buyers.
- bestsellng product
- 3 most profitable products and characteristics of our most loyal customers.


In [1]:
import pandas as pd


In [21]:
pur_behav=pd.read_csv("purchase_behaviour.csv")
trans_data=pd.read_csv("transaction_data.csv")

In [20]:
pur_behav.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 72637 entries, 0 to 72636
Data columns (total 3 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   LYLTY_CARD_NBR    72637 non-null  int64 
 1   LIFESTAGE         72637 non-null  object
 2   PREMIUM_CUSTOMER  72637 non-null  object
dtypes: int64(1), object(2)
memory usage: 1.7+ MB


In [7]:
trans_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 264836 entries, 0 to 264835
Data columns (total 8 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   DATE            264836 non-null  int64  
 1   STORE_NBR       264836 non-null  int64  
 2   LYLTY_CARD_NBR  264836 non-null  int64  
 3   TXN_ID          264836 non-null  int64  
 4   PROD_NBR        264836 non-null  int64  
 5   PROD_NAME       264836 non-null  object 
 6   PROD_QTY        264836 non-null  int64  
 7   TOT_SALES       264836 non-null  float64
dtypes: float64(1), int64(6), object(1)
memory usage: 16.2+ MB


In [8]:
pur_behav.head(5)

LYLTY_CARD_NBR               LIFESTAGE PREMIUM_CUSTOMER
0            1000   YOUNG SINGLES/COUPLES          Premium
1            1002   YOUNG SINGLES/COUPLES       Mainstream
2            1003          YOUNG FAMILIES           Budget
3            1004   OLDER SINGLES/COUPLES       Mainstream
4            1005  MIDAGE SINGLES/COUPLES       Mainstream

In [11]:
trans_data.head(5)

DATE  STORE_NBR  LYLTY_CARD_NBR  TXN_ID  PROD_NBR  \
0  43390          1            1000       1         5   
1  43599          1            1307     348        66   
2  43605          1            1343     383        61   
3  43329          2            2373     974        69   
4  43330          2            2426    1038       108   

                                  PROD_NAME  PROD_QTY  TOT_SALES  
0    Natural Chip        Compny SeaSalt175g         2        6.0  
1                  CCs Nacho Cheese    175g         3        6.3  
2    Smiths Crinkle Cut  Chips Chicken 170g         2        2.9  
3    Smiths Chip Thinly  S/Cream&Onion 175g         5       15.0  
4  Kettle Tortilla ChpsHny&Jlpno Chili 150g         3       13.8

In [12]:
trans_data

DATE  STORE_NBR  LYLTY_CARD_NBR  TXN_ID  PROD_NBR  \
0       43390          1            1000       1         5   
1       43599          1            1307     348        66   
2       43605          1            1343     383        61   
3       43329          2            2373     974        69   
4       43330          2            2426    1038       108   
...       ...        ...             ...     ...       ...   
264831  43533        272          272319  270088        89   
264832  43325        272          272358  270154        74   
264833  43410        272          272379  270187        51   
264834  43461        272          272379  270188        42   
264835  43365        272          272380  270189        74   

                                       PROD_NAME  PROD_QTY  TOT_SALES  
0         Natural Chip        Compny SeaSalt175g         2        6.0  
1                       CCs Nacho Cheese    175g         3        6.3  
2         Smiths Crinkle Cut  Chips Chicken 170g         2        2.9  
3         Smiths Chip Thinly  S/Cream&Onion 175g         5       15.0  
4       Kettle Tortilla ChpsHny&Jlpno Chili 150g         3       13.8  
...                                          ...       ...        ...  
264831   Kettle Sweet Chilli And Sour Cream 175g         2       10.8  
264832             Tostitos Splash Of  Lime 175g         1        4.4  
264833                  Doritos Mexicana    170g         2        8.8  
264834   Doritos Corn Chip Mexican Jalapeno 150g         2        7.8  
264835             Tostitos Splash Of  Lime 175g         2        8.8  

[264836 rows x 8 columns]

In [22]:
df=trans_data

In [23]:
customers = pd.read_csv("purchase_behaviour.csv")  # Customer segmentation data
transactions = pd.read_csv("transaction_data.csv")  # Transaction data

In [24]:
# Merge on LYLTY_CARD_NBR
df = transactions.merge(customers, on="LYLTY_CARD_NBR")

In [41]:
top_products = df.groupby("PROD_NAME")["TOT_SALES"].sum().sort_values(ascending=False)
top_products

PROD_NAME
Dorito Corn Chp     Supreme 380g            40352.0
Smiths Crnkle Chip  Orgnl Big Bag 380g      36367.6
Smiths Crinkle Chips Salt & Vinegar 330g    34804.2
Kettle Mozzarella   Basil & Pesto 175g      34457.4
Smiths Crinkle      Original 330g           34302.6
                                             ...   
WW Crinkle Cut      Chicken 175g             4702.2
Sunbites Whlegrn    Crisps Frch/Onin 90g     4600.2
WW Crinkle Cut      Original 175g            4532.2
Woolworths Mild     Salsa 300g               4234.5
Woolworths Medium   Salsa 300g               4050.0
Name: TOT_SALES, Length: 114, dtype: float64

In [25]:
# 1. Top 3 Most Profitable Products
top_products = df.groupby("PROD_NAME")["TOT_SALES"].sum().sort_values(ascending=False).head(3)
print("Top 3 Most Profitable Products:\n", top_products)


Top 3 Most Profitable Products:
 PROD_NAME
Dorito Corn Chp     Supreme 380g            40352.0
Smiths Crnkle Chip  Orgnl Big Bag 380g      36367.6
Smiths Crinkle Chips Salt & Vinegar 330g    34804.2
Name: TOT_SALES, dtype: float64


* Top 3 Most Profitable Products: *
- PROD_NAME
- Dorito Corn Chp     Supreme 380g            40352.0
- Smiths Crnkle Chip  Orgnl Big Bag 380g      36367.6
- Smiths Crinkle Chips Salt & Vinegar 330g    34804.2
- Name: TOT_SALES, dtype: float64

In [26]:
# 2. Bestselling Products (based on quantity sold)
bestselling_products = df.groupby("PROD_NAME")["PROD_QTY"].sum().sort_values(ascending=False).head(3)
print("Bestselling Products:\n", bestselling_products)


Bestselling Products:
 PROD_NAME
Dorito Corn Chp     Supreme 380g            6509
Kettle Mozzarella   Basil & Pesto 175g      6381
Kettle Tortilla ChpsHny&Jlpno Chili 150g    6309
Name: PROD_QTY, dtype: int64


* Bestselling Products: * 
 - PROD_NAME
- Dorito Corn Chp     Supreme 380g            6509
- Kettle Mozzarella   Basil & Pesto 175g      6381
- Kettle Tortilla ChpsHny&Jlpno Chili 150g    6309
- Name: PROD_QTY, dtype: int64

In [28]:
# 3. Identifying Loyal Customers
loyal_customers = df.groupby("LYLTY_CARD_NBR").agg(
    total_spend=("TOT_SALES", "sum"),
    num_transactions=("TXN_ID", "count"),
    avg_order_value=("TOT_SALES", "mean")
)
loyal_customers = loyal_customers.merge(customers, on="LYLTY_CARD_NBR")

# Analyzing loyalty trends
loyalty_analysis = loyal_customers.groupby(["LIFESTAGE", "PREMIUM_CUSTOMER"]).agg(
    avg_total_spend=("total_spend", "mean"),
    avg_num_transactions=("num_transactions", "mean")
).reset_index()

print("Loyal Customer Characteristics:\n", loyalty_analysis)


Loyal Customer Characteristics:
                  LIFESTAGE PREMIUM_CUSTOMER  avg_total_spend  \
0   MIDAGE SINGLES/COUPLES           Budget        23.613564   
1   MIDAGE SINGLES/COUPLES       Mainstream        27.186781   
2   MIDAGE SINGLES/COUPLES          Premium        24.036466   
3             NEW FAMILIES           Budget        19.719829   
4             NEW FAMILIES       Mainstream        20.039929   
5             NEW FAMILIES          Premium        19.542687   
6           OLDER FAMILIES           Budget        36.013529   
7           OLDER FAMILIES       Mainstream        36.540286   
8           OLDER FAMILIES          Premium        36.041513   
9    OLDER SINGLES/COUPLES           Budget        27.747981   
10   OLDER SINGLES/COUPLES       Mainstream        27.057566   
11   OLDER SINGLES/COUPLES          Premium        27.844874   
12                RETIREES           Budget        25.403637   
13                RETIREES       Mainstream        24.027944   
14     